src/output.cpp

```diff
@@ -1081,6 +1271,93 @@ void JsonOutput::map_hist(
   out_ << "}}" << std::endl;
 }
 
+void JsonOutput::map_tseries(
+    BPFtrace &bpftrace,
+    const BpfMap &map,
+    const TSeriesMap &values_by_key,
+    const std::vector<std::pair<KeyType, EpochType>> &latest_epoch_by_key) const
+{
+  const auto &map_info = bpftrace.resources.maps_info.at(map.name());
+
+  out_ << R"({"type": ")" << MessageType::tseries << R"(", "data": {)";
+  out_ << "\"" << json_escape(map.name()) << "\": ";
+  if (!map_info.is_scalar) // check if this map has keys
```

Comment:
Probably dumb question: where the logic is if the map is not a scalar?

Reply:
Non-scalar maps need the enclosing `{` `}`, since it contains a map of key names to data points instead of just a single set of data points.

**Scalar**
```
{"type": "tseries", "data": {"@": [{"interval_start":"2025-07-08T00:06:31.921000000Z","value":1}]}}
```

**Non Scalar**
```
{"type": "tseries", "data": {"@": {"1": [{"interval_start":"2025-07-08T00:06:24.989000000Z","value":1}]}}}
```

I think there is some similar looking code in the lhist JSON output code as well.

Reply:
Ahh I see. Thanks!